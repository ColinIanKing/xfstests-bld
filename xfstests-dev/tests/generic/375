#! /bin/bash
# FS QA Test 375
#
# Check if SGID is cleared upon chmod / setfacl when the owner is not in the
# owning group.
#
#-----------------------------------------------------------------------
# Copyright (c) 2016 Red Hat.  All Rights Reserved.
#
# Author: Andreas gruenbacher <agruenba@redhat.com>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it would be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write the Free Software Foundation,
# Inc.,  51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
#-----------------------------------------------------------------------
#

seq=`basename $0`
seqres=$RESULT_DIR/$seq
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1	# failure is the default!
trap "_cleanup; exit \$status" 0 1 2 3 15

_cleanup()
{
	cd /
	rm -f $tmp.*
}

# get standard environment, filters and checks
. ./common/rc
. ./common/filter
. ./common/attr

# real QA test starts here
_supported_fs generic
_supported_os Linux
_require_test
_require_runas
_require_acls

cd $TEST_DIR
testfile=testfile.$seq
rm -f $testfile

touch $testfile
chown 100:100 $testfile

echo '*** SGID should remain set (twice)'
chmod 2755 $testfile
_runas -u 100 -g 100 -- chmod 2777 $testfile
stat -c %A $testfile
chmod 2755 $testfile
_runas -u 100 -g 100 -- setfacl -m u::rwx,g::rwx,o::rwx $testfile
stat -c %A $testfile

echo '*** SGID should be cleared (twice)'
chmod 2755 $testfile
_runas -u 100 -g 101 -- chmod 2777 $testfile
stat -c %A $testfile
chmod 2755 $testfile
_runas -u 100 -g 101 -- setfacl -m u::rwx,g::rwx,o::rwx $testfile
stat -c %A $testfile

echo '*** Expected failure'
_runas -u 101 -g 101 -- chmod 2777 $testfile

status=0
exit
