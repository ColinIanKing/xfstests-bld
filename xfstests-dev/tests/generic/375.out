QA output created by 375
*** SGID should remain set (twice)
-rwxrwsrwx
-rwxrwsrwx
*** SGID should be cleared (twice)
-rwxrwxrwx
-rwxrwxrwx
*** Expected failure
chmod: changing permissions of 'testfile.375': Operation not permitted
